docs/proposals/node-allocatable.md

```diff
@@ -0,0 +1,155 @@
+   workloads. For example, processes scheduled to run alongside Kubernetes on Mesos.
+3. **Kubelet Allocatable** - Compute resources available for scheduling (including scheduled &
+   unscheduled resources). This value is currently implicit, and the focus of this proposal. See
+   [below](#api-changes) for more details.
+4. **System Reserved** (proposed) - Compute resources reserved for system components such as the docker daemon, kubelet, kube proxy, etc.
+
+### API changes
+
+#### Allocatable
+
+Add `Allocatable` (4) to
+[`NodeStatus`](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/HEAD/docs/api-reference/v1/definitions.html#_v1_nodestatus):
+
+```
+type NodeStatus struct {
+  ...
+  // Allocatable represents schedulable resources of a node.
+  Allocatable ResourceList `json:"allocatable,omitempty"`
+  ...
+}
+```
+
+Allocatable will be computed by the Kubelet and reported periodically to the API server. It is
+defined to be:
+
+```
+   [Allocatable] = [Machine Capacity] - ( [Reserved] + [System Reserved] + [Kernel usage (smoothed)] )
+```
+
+The scheduler will use `Allocatable` in place of `Capacity` when scheduling pods, and the Kubelet
+will use it when perming admission checks.
+
+- Since kernel usage can fluctuate and is out of kubernetes control, `Allocatable` may vary over
+  time.
+- Scheduler can compute unscheduled resources as `[Allocatable] - [Usage]`, where usage is gathered
+  from the [kubelet metrics API](compute-resource-metrics-api.md).
+
+#### System Reserved
+
+- **TODO**: Determine which resources *must* be specified (CPU+memory?)
+- **TODO**: Decide on format (parsed string vs. individual resource flags)
+
+`SystemReserved` is a **required** parameter specifying resources reserved for system components
+(4). It is provided as a command-line flag to the Kubelet at startup, and therefore cannot be
+changed during normal Kubelet operation (this could change in the [future](#future-work)).
+
+#### Reserved
+
+`Reserved` is an **optional** parameter specifying reserved resources (2). It is specified as a
+Kubelet command line flag using the same format as [system reserved](#system-reserved).
+
+## Issues
+
+### System reservation too small
+
+Solution **TBD**
+
+**Option 1:** Do nothing (best effort). Let the system daemons overflow the reserved resources and
+  hope for the best. If the machine usage is less than Allocatable, there will be some room for
+  overflow and the node should continue to function. If the machine has been scheduled to capacity
+  (worst-case scenario), it may enter an unstable state.
+
+**Option 2:** Assume the worst and transition the node to the not-ready state. The kubelet could
+  attempt to reduce usage by evicting pods until usage comes under control. At that point the node
+  could be made available again.
+
+### Version discrepency
+
+**API server / scheduler is not allocatable-resources aware:** The kubelet already performs
```

Comment:
This scenario is actually pretty bad because a scheduler that is still using Capacity to schedule may get into an infinite loop scheduling a Pod onto a node that Kubelet keeps rejecting because Kubelet is using Allocatable to determine feasibility, and Allocatable < Capacity. Maybe we should roll this out in two steps, first start reporting Allocatable but still have Kubelet use Capacity for feasibility check, then later once people have had a chance to update their schedulers, switch Kubelet feasibility check to use Allocatable. This may allow the machine to be overcommitted until the second step, but it will be no worse than today.